packages/tangle/message.go

```diff
@@ -838,6 +850,7 @@ func (m *MessageMetadata) SetScheduled(scheduled bool) (modified bool) {
 	}
 
 	m.scheduled = scheduled
+	m.scheduledTime = clock.SyncedTime()
```

Comment:
Either you need to grab `scheduledTimeMutex` here or use `scheduledMutex` also for accessing `ScheduledTime()`